TheJumpCloud/jumpcloud-ADMU
# windows 10 fix for start menu/ search

## Issues
* [SA-3524](https://jumpcloud.atlassian.net/browse/3524) - Fix for windows 10 start menu / search bar

## What does this solve?

The windows 10 search and start menu applications are based on the universal windows platform application (uwp) package language. These apps need to be forcefully re-registered to a new local user when migrating. This used to be accomplished through the jcadmu_uwp.exe app (the first run package when the user first logs in) however at appears that a change in Windows 10 has rendered our old fix no longer viable. 

In this branch, after copying the user registry hive from the AD user, to the new local user registry hive, the following keys are deleted:

```pwsh
"HKEY_USERS:\NewLocalUser\SOFTWARE\Microsoft\Windows\CurrentVersion\StartLayout"
"HKEY_USERS:\NewLocalUser\SOFTWARE\Microsoft\Windows\CurrentVersion\Start"
"HKEY_USERS:\NewLocalUser\SOFTWARE\Microsoft\Windows\CurrentVersion\SearchSettings"
"HKEY_USERS:\NewLocalUser\SOFTWARE\Microsoft\Windows\CurrentVersion\Search"
```
After migration the start menu and search applications are forcefully re-registered to the new local user (presumably through a built-in windows process since the keys are automatically re-created). The start menu should work and the search menu should function as expected.

## Is there anything particularly tricky?

## How should this be tested?

On a Windows 10 device:

- Migrate an AzureAD or on-prem AD user
- The ADMU log should log: `Windows 10 System, removing start and search reg keys to force refresh of those apps`
- Upon login to the migrated user, the start menu and search bar should function as expected

On a Windows 11 device:

- Migrate an AzureAD or on-prem AD user
- The ADMU log should NOT log: `Windows 10 System, removing start and search reg keys to force refresh of those apps`
- Upon login to the migrated user, the start menu and search bar should function as expected

## Screenshots

[SA-3524]: https://jumpcloud.atlassian.net/browse/SA-3524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
